Update generators to work with newest table structure

bdach opened the pull request:
See https://github.com/ppy/osu-queue-score-statistics/pull/198. C# class definitions lifted verbatim from therein.

PRing early as draft as I haven't fully tested this yet. I have had a PP sheet complete, so I have reasonable confidence that it'll work; the score sheet only works in my head for now. Will undraft once I have more confidence that it works.

bdach committed:
Update definition of `solo_scores` table

bdach committed:
Update sheet generators in line with new database structure

bdach committed:
Remove usages of no-longer-existing helper tables

smoogipoo reviewed (approved):
I'm reasonably confident this works, ready to merge when you pull it out of draft

bdach committed:
Remove stray closing bracket

bdach committed:
Fix scoring query failing to subtract two `unsigned int` values

bdach committed:
Update game packages

Targeted at runtime failures such as:

	generator-1  | Unhandled exception. System.Data.DataException: Error parsing column 12 (data={"mods": [{"acronym": "DT"}, {"acronym": "FL"}, {"acronym": "HR"}, {"acronym": "HD"}, {"acronym": "NF"}, {"acronym": "CL"}], "statistics": {"meh": 1, "miss": 10, "great": 1}, "maximum_statistics": {"great": 12, "legacy_combo_increase": 185}} - String)
	generator-1  |  ---> Newtonsoft.Json.JsonSerializationException: Could not convert string 'legacy_combo_increase' to dictionary key type 'osu.Game.Rulesets.Scoring.HitResult'. Create a TypeConverter to convert from the string to the key type object. Path 'maximum_statistics.legacy_combo_increase', line 1, position 235.
	generator-1  |  ---> System.ArgumentException: Requested value 'legacy_combo_increase' was not found.
	generator-1  |    at Newtonsoft.Json.Utilities.EnumUtils.ParseEnum(Type enumType, NamingStrategy namingStrategy, String value, Boolean disallowNumber)
	generator-1  |    at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.PopulateDictionary(IDictionary dictionary, JsonReader reader, JsonDictionaryContract contract, JsonProperty containerProperty, String id)

The previous packages were several years out of date at this point, and
`legacy_combo_increase` indeed did not exist in those better days...

bdach commented:
There were a few extra changes required to make the scoring sheet work, see commit messages for rationale for each. https://github.com/smoogipoo/diffcalc-sheet-generator/pull/1/commits/3a31521d27203b57e6ddd22df2ea94e36890fdce decided to be a problem only now because the previous `SoloScoreInfo` structure did not have `MaximumStatistics` inside... Can't tell if that's good or bad.

Other than that, should be ready to go now, I think?
